specifyweb/frontend/js_src/lib/components/TreeView/CreateTree.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ *WIP File to separate out functions for adding trees.
+ *Contains stuff for dialogs, etc.
+ *
+ *TODO:
+ *- Add localizations for the tree default names
+ *- Add page reload upon save
+ *- Remove comments
+ */
+import React from 'react';
+
+import { commonText } from '../../localization/common';
+import { treeText } from '../../localization/tree';
+import { DeepPartial } from '../../utils/types';
+import { Ul } from '../Atoms';
+import { Button } from '../Atoms/Button';
+import { className } from '../Atoms/className';
+import type { AnySchema, AnyTree , SerializedResource} from '../DataModel/helperTypes';
+import { SpecifyResource } from '../DataModel/legacyTypes';
+import { deserializeResource } from '../DataModel/serializers';
+import { TaxonTreeDef } from '../DataModel/types';
+import { ResourceView } from '../Forms/ResourceView';
+import { userInformation } from '../InitialContext/userInformation';
+import { Dialog } from '../Molecules/Dialog';
+import { defaultTreeDefs } from './TreeDefaults';
+
+export function CreateTree<SCHEMA extends AnyTree>({
+  tableName,
+}: {
+  readonly tableName: SCHEMA['tableName'];
+}): JSX.Element {
+
+  const [isActive, setIsActive] = React.useState(0);
+  const [selectedResource, setSelectedResource] = React.useState<SpecifyResource<AnySchema> | null>(null);
```

Comment:
you used `null` in this file in several places?
could you please use `undefined` instead?
that would be more consistent with what we use in the rest of the codebase.
thanks 😁

Reply:
That makes sense for the highlighted section w/ selectedResource, but `null` is used for false conditions in a few places in Tree.tsx and Actions.tsx. I'm pretty sure it's fine in the rest of the file since it's just to say "don't render anything"